compiler/src/dotty/tools/dotc/sbt/ExtractAPI.scala

```diff
@@ -212,8 +212,7 @@ private class ExtractAPICollector(implicit val ctx: Context) extends ThunkHolder
 
     // Synthetic methods that are always present do not affect the API
     // and can therefore be ignored.
-    def alwaysPresent(s: Symbol) =
-      s.isCompanionMethod || (csym.is(ModuleClass) && s.isConstructor)
+    def alwaysPresent(s: Symbol) = csym.is(ModuleClass) && s.isConstructor
```

Comment:
Why don't declarations in the API extractor include companion methods? What invariant has opaque types introduced to make the previous code obsolete?

Reply:
Companion methods don't exist anymore. (They were the synthetic methods that pointed to the companion class or object to the other. We now use an explicit field in `ClassDenotation` for this).